### PR TITLE
layout: Make `position: relative` with non-auto `z-index` create a stacking context.

### PR DESCRIPTION
**Pull request by pcwalton**
Improves imgur.com and Fast Company articles.

This change made `min-height-106.htm.ini` fail because the thing it was testing
for never worked: we were relying on the incorrect stacking order of `position:
relative` to get the green square to show up.

r? @glennw

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7277)

### CODE REVIEW

File: components/layout/fragment.rs [hunk 1]
```diff
@@ -2013,10 +2013,14 @@ impl Fragment {
...
+             overflow_x::T::visible,
              overflow_x::T::visible) => false,
             (position::T::absolute, _, _, _) |
-            (position::T::fixed, _, _, _) => true,
-            (position::T::relative, _, _, _) |
+            (position::T::fixed, _, _, _) |
+            (position::T::relative, _, _, _) => true,
             (position::T::static_, _, _, _) => {
                 // FIXME(pcwalton): `position: relative` establishes a new stacking context if
                 // `z-index` is not `auto`. But this matches what we did before.
```
Comment (samlh) [inline components/layout/fragment.rs]: Is this comment still accurate?

File: tests/wpt/mozilla/tests/mozilla/canvas/drawimage_html_image_5_ref.html [hunk 2]
```diff
@@ -20,7 +20,6 @@
     left: 50px;
     width: 64px;
     height: 32px;
-    overflow: hidden;
```
Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/canvas/drawimage_html_image_5_ref.html]: Why this change?

Comment (pcwalton) [inline tests/wpt/mozilla/tests/mozilla/canvas/drawimage_html_image_5_ref.html]: There was a pre-existing bug whereby stacking contexts can get incorrectly clipped.
